CONTRIBUTING.md

```diff
@@ -1,6 +1,143 @@
-# Contributing guidelines
+# Contributing to GLASS
 
-## Workflow
+If you are planning to develop _GLASS_, or want to use the latest commit of
+_GLASS_ on your local machine, you might want to install it from the source.
+This installation is not recommended for users who want to use the stable
+version of _GLASS_. The page below describes how to build, test, and develop
+_GLASS_.
+
+## Installation
+
+The developer installation of _GLASS_ comes with several optional dependencies -
+
+- `test`: installs `scipy`, `fitsio`, and the relevant `pytest` dependencies
+- `docs`: installs `sphinx`, `nbsphinx`, and a few other docs related
+  dependencies
+- `examples`: installs `camb`, `glass.ext.camb`, and a few notebook related
+  dependencies
+
+These options can be used with `pip` with the editable (`-e`) mode of
+installation in the following way -
+
+```bash
+pip install -e .[docs,test]
+```
+
+## Tooling
+
+### Pre-commit
+
+_GLASS_ uses a set of `pre-commit` hooks and the `pre-commit.ci` bot to format,
+lint, and prettify the codebase. The hooks can be installed locally using -
+
+```bash
+pre-commit install
+```
+
+This would run the checks every time a commit is created locally. The checks
+will only run on the files modified by that commit, but the checks can be
+triggered for all the files using -
+
+```bash
+pre-commit run --all-files
+```
+
+If you would like to skip the failing checks and push the code for further
+discussion, use the `--no-verify` option with `git commit`.
+
+## Testing
+
+_GLASS_ is tested using `pytest` and `pytest-doctestplus`. `pytest` is
+responsible for testing the code, whose configuration is available in
+[pyproject.toml](https://github.com/glass-dev/glass/blob/main/pyproject.toml).
+`pytest-doctestplus` is responsible for testing the examples available in every
+docstring, which prevents them from going stale. Additionally, _GLASS_ also uses
+`pytest-cov` (and [Coveralls](https://coveralls.io)) to calculate/display the
+coverage of these unit tests.
+
+### Running tests locally
+
+The tests can be executed using the `test` dependencies of _GLASS_ in the
+following way -
+
+```bash
+python -m pytest --cov=glass --doctest-plus -v
```

Comment:
For another PR, but we can configure the coverage in `pyproject.toml` so one doesn't need to explicitly write `--cov=glass`. https://github.com/astro-informatics/sleplet/blob/782279ec1be404aa356c6e0b583bd536cf3a4789/pyproject.toml#L101-L109

Reply:
Added `-v` in the recent PR. I usually avoid adding the `cov` and the `doctest-plus` option in pyproject.toml because we don't need those always.

Reply:
I agree, my suggestion is you add to it the `coverage` section. So when you ask for coverage you get it on `glass` without having to specify it - I'm not asking for the `pytest` section

Reply:
Ahh, let me quickly check that.

Reply:
Don't worry we can fix in another issue https://github.com/glass-dev/glass/issues/282